tree/treeplayer/src/TTreeIndex.cxx

```diff
@@ -638,3 +638,50 @@ void TTreeIndex::SetTree(TTree *T)
    fTree = T;
 }
 
+/// \cond HIDDEN_SYMBOLS
+////////////////////////////////////////////////////////////////////////////////
+/// \brief Create a deep copy of the TTreeIndex
+/// The new index is not attached to any tree. It is the responsibility of the
+/// caller to attach it to a tree if necessary.
+std::unique_ptr<TVirtualIndex> TTreeIndex::MakeIndexCopy() const
+{
+   auto index = std::make_unique<TTreeIndex>();
+
+   if (fN == 0)
+      return index;
+
+   index->fMajorName = fMajorName;
+   index->fMinorName = fMinorName;
+   index->fN = fN;
+
+   if (!fIndexValues) {
+      index->fIndexValues = nullptr;
+   } else {
+      index->fIndexValues = new Long64_t[index->fN];
+      std::copy(fIndexValues, fIndexValues + fN, index->fIndexValues);
+   }
+
+   if (!fIndexValuesMinor) {
+      index->fIndexValuesMinor = nullptr;
+   } else {
+      index->fIndexValuesMinor = new Long64_t[index->fN];
+      std::copy(fIndexValuesMinor, fIndexValuesMinor + fN, index->fIndexValuesMinor);
+   }
+
+   if (!fIndex) {
+      index->fIndex = nullptr;
+   } else {
+      index->fIndex = new Long64_t[index->fN];
+      std::copy(fIndex, fIndex + fN, index->fIndex);
+   }
+
+   index->fMajorFormula = (fMajorFormula ? static_cast<TTreeFormula *>(fMajorFormula->Clone()) : nullptr);
+   index->fMinorFormula = (fMinorFormula ? static_cast<TTreeFormula *>(fMinorFormula->Clone()) : nullptr);
+   index->fMajorFormulaParent =
+      (fMajorFormulaParent ? static_cast<TTreeFormula *>(fMajorFormulaParent->Clone()) : nullptr);
+   index->fMinorFormulaParent =
+      (fMinorFormulaParent ? static_cast<TTreeFormula *>(fMinorFormulaParent->Clone()) : nullptr);
```

Comment:
Why though?

Reply:
I think I understand your point, it's a question of semantics. Since the function has the word `Copy` in the name, it should behave as close as possible as a copy-constructor (although we don't provide the copy-constructor explicitly to avoid unexpected/unwanted copies). So the only "optimization" that I can do is `if (fN ==0)` I don't bother creating the arrays data members. Everything else should still be copied as it would happen in a copy constructor. Is that the reasoning here?

Reply:
In practical term, it should end up with a result similar to what `Clone` is doing.   In term of expectation, one would expect that the 'Copy' is in a valid state.

So it should updates "only":

```
     fMajorName
     fMinorName
     fN;
     fIndexValues 
     fIndexValuesMinor 
     fIndex
```
and that's enough.